lib/bokken/accounts/user_skill.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Bokken.Accounts.UserSkill do
+  @moduledoc """
+  User (ninja / mentor) and skill join table
+  """
+  use Bokken.Schema
+
+  alias Bokken.Accounts.{Mentor, Ninja, Skill}
+
+  @required_fields [:skill_id]
+  @optional_fields [:mentor_id, :ninja_id]
+
+  schema "user_skills" do
+    belongs_to :skill, Skill, type: :binary_id
+    belongs_to :mentor, Mentor, type: :binary_id
+    belongs_to :ninja, Ninja, type: :binary_id
```

Comment:
```suggestion
    belongs_to :ninja, Ninja
```

No need to specify the type of the foreign_key because it is done in `Bokken.Schema`.